styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -101,7 +103,6 @@ public void initialize(RunState runState) throws IsClosed {
     final long counter;
     try {
       counter = storage.getLatestStoredCounter(workflowInstance).orElse(NO_EVENTS_PROCESSED);
-      storeActivation(workflowInstance, counter);
```

Comment:
This would just have stored whatever was the last counter value (or -1) in datastore, which then was incremented further down in the actual state.

Since there's no read-increment-write sequence in `process()`, but rather a `currentCount = counter++`, this will behave the same (that is, it will write events starting at 0,1,2,... or prevTrig+1, prevTrig+2...).